homeassistant/components/vacuum/device_trigger.py

```diff
@@ -0,0 +1,86 @@
+"""Provides device automations for Vacuum."""
+from typing import List
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_DOMAIN,
+    CONF_TYPE,
+    CONF_PLATFORM,
+    CONF_DEVICE_ID,
+    CONF_ENTITY_ID,
+)
+from homeassistant.core import HomeAssistant, CALLBACK_TYPE
+from homeassistant.helpers import config_validation as cv, entity_registry
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.components.automation import state, AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from . import DOMAIN, STATE_CLEANING, STATE_DOCKED, STATES
+
+TRIGGER_TYPES = {"cleaning", "docked"}
```

Comment:
This is only for the new vacuum devices, the old one should be removed (and should have been removed when we introduced the new one)